pkg/sfu/buffer/rtpstats.go

```diff
@@ -1288,24 +1329,33 @@ func (r *RTPStats) updateGapHistogram(gap int) {
 	}
 }
 
-func (r *RTPStats) getAndResetSnapshot(snapshotId uint32) (*Snapshot, *Snapshot) {
+func (r *RTPStats) getAndResetSnapshot(snapshotId uint32, override bool) (*Snapshot, *Snapshot) {
 	if !r.initialized || (r.params.IsReceiverReportDriven && r.lastRRTime.IsZero()) {
 		return nil, nil
 	}
 
 	then := r.snapshots[snapshotId]
 	if then == nil {
 		then = &Snapshot{
-			startTime:  r.startTime,
-			extStartSN: r.extStartSN,
+			startTime:            r.startTime,
+			extStartSN:           r.extStartSN,
+			extStartSNOverridden: r.extStartSN,
 		}
 		r.snapshots[snapshotId] = then
 	}
 
+	var startTime time.Time
+	if override && r.params.IsReceiverReportDriven {
+		startTime = r.lastRRTime
```

Comment:
Lock snap shots to receiver report time. Without that, maxPPS calculation was off as the window durations were offset between receiver report and snap shots.